src/test/ui/nll/mir_check_cast_reify.stderr

```diff
@@ -4,14 +4,14 @@ warning: not reporting region error due to nll
 LL |     let f: fn(_) -> _ = foo;
    |                         ^^^
 
-error: borrowed data escapes outside of function
-  --> $DIR/mir_check_cast_reify.rs:48:5
+error: unsatisfied lifetime constraints
+  --> $DIR/mir_check_cast_reify.rs:46:25
    |
 LL | fn bar<'a>(x: &'a u32) -> &'static u32 {
-   |            - `x` is a reference that is only valid in the function body
+   |        -- lifetime `'a` defined here
 ...
-LL |     f(x)
-   |     ^^^^ `x` escapes the function body here
+LL |     let f: fn(_) -> _ = foo;
+   |                         ^^^ cast requires that `'a` must outlive `'static`
```

Comment:
Hmm, these errors *do* look less good, but they are also not *wrong*; I certainly see how they arise.

Reply:
This is a tricky case, but I think if we provide the suggested output in the other comment it would be slightly more understandable. Something closer to the ideal output in my mind would be:

```
error: unsatisfied lifetime constraints
  --> $DIR/mir_check_cast_reify.rs:46:25
   |
LL | fn foo<'a>(x: &'a u32) -> &'a u32
   | --------------------------------- `foo` is not `'static` due to this definition
...
LL | fn bar<'a>(x: &'a u32) -> &'static u32 {
   |        --     ---         -------- `foo` needs to be `'static` due to this return type
   |        |      |
   |        |      `x` is `'a` due to this
   |        lifetime `'a` defined here
...
LL |     let f: fn(_) -> _ = foo;
   |                         ^^^ this is `'a` but it must outlive `'static`
```